contracts/test/StakeTable.t.sol

```diff
@@ -637,6 +656,596 @@ contract StakeTable_register_Test is LightClientCommonTest {
+        vm.expectRevert(abi.encodeWithSelector(S.InsufficientBalance.selector, 0));
+        stakeTable.undelegate(validator, INITIAL_BALANCE);
+        vm.stopPrank();
+    }
+
+    function test_WithdrawalAfterValidatorExit() public {
+        // Should test withdrawing after validator exit
+        vm.prank(tokenGrantRecipient);
+        token.transfer(delegator, INITIAL_BALANCE);
+
+        registerValidatorOnStakeTable(validator, seed1, COMMISSION, stakeTable);
+
+        vm.startPrank(delegator);
+        token.approve(address(stakeTable), INITIAL_BALANCE);
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.Delegated(delegator, validator, INITIAL_BALANCE);
+        stakeTable.delegate(validator, INITIAL_BALANCE);
+        vm.stopPrank();
+
+        vm.startPrank(validator);
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.ValidatorExit(validator);
+        stakeTable.deregisterValidator();
+        vm.stopPrank();
+        (uint256 validatorAmountDelegated,) = stakeTable.validators(validator);
+        assertEq(validatorAmountDelegated, 0);
+        (uint256 delegatorAmountUndelegated, uint256 unlocksAt) =
+            stakeTable.undelegations(validator, delegator);
+        assertEq(delegatorAmountUndelegated, 0); // undelegated amount is zero because the user did
+            // not undelegate
+        assertEq(unlocksAt, 0); // unlocks at is zero because the user did not undelegate
+
+        vm.startPrank(delegator);
+        vm.expectRevert(abi.encodeWithSelector(S.ValidatorAlreadyExited.selector));
+        stakeTable.undelegate(validator, INITIAL_BALANCE);
+
+        //validator already exited so call claim withdaral exit
+        vm.warp(block.timestamp + ESCROW_PERIOD);
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.Withdrawal(delegator, INITIAL_BALANCE);
+        stakeTable.claimValidatorExit(validator);
+        vm.stopPrank();
+
+        assertEq(token.balanceOf(delegator), INITIAL_BALANCE);
+        assertEq(token.balanceOf(address(stakeTable)), 0);
+        (validatorAmountDelegated,) = stakeTable.validators(validator);
+        assertEq(validatorAmountDelegated, 0);
+        uint256 delegatedAmount = stakeTable.delegations(validator, delegator);
+        assertEq(delegatedAmount, 0);
+    }
+
+    function test_ValidatorExitDuringUndelegationPeriod() public {
+        // Should test withdrawing after validator exit
+        vm.prank(tokenGrantRecipient);
+        token.transfer(delegator, INITIAL_BALANCE);
+
+        registerValidatorOnStakeTable(validator, seed1, COMMISSION, stakeTable);
+
+        vm.startPrank(delegator);
+        token.approve(address(stakeTable), INITIAL_BALANCE);
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.Delegated(delegator, validator, INITIAL_BALANCE);
+        stakeTable.delegate(validator, INITIAL_BALANCE);
```

Comment:
I think it would  become a bit easier to read if we made a helper function for this like with the validator registration, since it's used quite a lot. Can be done as follow up work though.